src/schema/slack/functions/_scripts/src/templates/template_function.ts

```diff
@@ -0,0 +1,152 @@
+import { AllowedTypeValueObject } from "./types.ts";
+import SchemaTypes from "../../../../../schema_types.ts";
+import SlackSchemaTypes from "../../../../schema_types.ts";
+import { InternalSlackTypes } from "../../../../types/custom/mod.ts";
+import {
+  FunctionParameter,
+  FunctionProperties,
+  FunctionProperty,
+  FunctionRecord,
+} from "../types.ts";
+import {
+  autogeneratedComment,
+  getParameterType,
+  getSlackCallbackId,
+  renderTypeImports,
+} from "./template_utils.ts";
+
+const typeMap: Record<string, AllowedTypeValueObject> = {
+  SchemaTypes,
+  SlackTypes: SlackSchemaTypes,
+  InternalSlackTypes,
+};
+
+const schemaTypeMap = Object.entries(typeMap).reduce<AllowedTypeValueObject>(
+  (acc, [schemaKey, schemaTypes]) => {
+    for (const typeKey in schemaTypes) {
+      const val = schemaTypes[typeKey];
+      if (val instanceof Object) {
+        acc[val.id] = `${schemaKey}.${typeKey}`;
+      } else {
+        acc[val] = `${schemaKey}.${typeKey}`;
+      }
+    }
+    return acc;
+  },
+  {},
+);
+
+const propertyToTypeScript = (
+  property: FunctionProperty,
+): string => {
+  const typescript = [];
+  typescript.push(
+    `type: ${schemaTypeMap[getParameterType(property.type)]}`,
+  );
+  if (property.description) {
+    typescript.push(`description: "${property.description}"`);
+  }
+  if (property.title) {
+    typescript.push(`title: "${property.title}"`);
+  }
+  if ("items" in property) {
+    typescript.push(`items: ${propertyToTypeScript(property.items)}`);
+  }
+  if ("properties" in property) {
```

Comment:
Here this does not work because my property is of type `FunctionProperty` and I'm [using narrowing](https://www.typescriptlang.org/docs/handbook/2/narrowing.html) (Fil showed me this yesterday) to access the fields